src/lib/Microsoft.Health.Events/EventConsumers/Service/EventConsumerService.cs

```diff
@@ -6,24 +6,40 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net.Http;
 using System.Threading.Tasks;
+using Azure;
+using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using EnsureThat;
 using Microsoft.Health.Events.Model;
 using Microsoft.Health.Logging.Telemetry;
+using Polly;
 
 namespace Microsoft.Health.Events.EventConsumers.Service
 {
     public class EventConsumerService : IEventConsumerService
```

Comment:
This also impacts the Normalization to FHIR logic.  That has a separate mechanism for non-recoverable errors.  This would fail to retry on 429 for example due to throttling from the FHIR service.

Reply:
To avoid breaking the FHIR conversion logic we should make this optional.

Reply:
We currently retry on `HttpExceptions`. Wouldn't that handle the 429 scenario?

Reply:
I see the custom exception handling now: https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs. Let me update this to make it optional.

Reply:
I don't know if it comes back as an HTTP request exception or some other type.  The FHIR logic takes the opposite approach, we retry all exceptions except the ones we explicitly capture and emit telemetry for.  With out further analysis I am concerned this could introduce unintended side effects for the FHIR conversion. For now I think it would be safe to only apply this during normalization and we can have a work item to unify methods if appropriate in the future.